l10n_it_fatturapa_out/wizard/wizard_export_fatturapa.py

```diff
@@ -62,6 +62,12 @@ class WizardExportFatturapa(models.TransientModel):
     _name = "wizard.export.fatturapa"
     _description = "Export E-invoice"
 
+    @staticmethod
+    def encode(string_to_encode, max_chars, encoding='latin'):
+        res = string_to_encode.replace('\t', ' ').encode(
```

Comment:
Quanto segue, non vuole essere un qualcosa di bloccante ma, semplicemente, un piccolo *"hint"*... Uno spunto!  
Stavamo giusto parlando in ufficio dei `\t` e di tutti gli altri caratteri "whitespace" e ci sei venuto in mente...

Forse è addirittura *over-engineered* per questo caso specifico... Ma te la propongo comunque!

```suggestion
        import re

        regex = re.compile(r'\s+')
        res = regex.sub(' ', string_to_encode).encode(
```

---

Il carattere `\s` in un'espressione regolare, seleziona tutta una serie di caratteri "whitespace" (`\r`, `\n`, `\t`, `\f`, `\v` e ` `) e, in questo caso, con il simbolo `+` seleziona tutti questi caratteri presenti 1 o più volte consecutivamente...

In parole povere, un replace così eseguito, sostituisce tutti i caratteri "whitespace" consecutivi sempre e solo con un unico singolo spazio.  
È anche vero che, sostituisce lo spazio stesso con un altro spazio (proprio perché considerato carattere "whitespace") ma, in fin dei conti, rimuove tutta una serie di grane che, altrimenti, dovremmo gestire (eventualmente) indicando carattere per carattere...

---

Vedi tu se preferisci integrare o lasciare la PR così com'è! 😉

Reply:
:+1: